docs/whats_new_40.rst

```diff
@@ -0,0 +1,135 @@
+What's New in ReFrame 4.0
+=========================
+
+ReFrame 4.0 introduces some important new features and removes all features, configuration options and interfaces that were deprecated in the 3.x versions.
+It also introduces a couple of new deprecations.
+
+ReFrame 4.0 maintains backward compatibility as much as possible.
+Existing 3.x configurations and 3.x tests are expected to run out-of-the-box, despite any warnings issued.
+Framework's behavior with respect to performance logging has also changed, but configuration options are offered so that users can switch to the old behavior.
```

Comment:
```suggestion
The framework's behavior with respect to performance logging has also changed, but configuration options are offered so that users can switch to the old behavior.
```